### ISSUE TITLE
add local chat history deletion

### ISSUE THREAD

**Issue by fu3x11** [tigase/siskin-im#240]
and allow them to expire from the device

**Comment by hantu85** [comment]
That is not the same and I do not suppose this would be expected result of chat history removal operation. Anyway, that would require some work and it would need to be part of the future releases as I've stated.

**Comment by licaon-kter** [comment]
> That is not the same and I do not suppose this would be expected result of chat history removal operation.

Not sure a follow, in a federated system we should not expect the history to disappear from MAM or other devices.

While that is true, I see no issue with replacing current local DB entries with "NIL" and keep the rest as it is, so there's no reason to pull from MAM and re-get the same messages because the DB will always say "I'm up to date"

But the user get what they want, if they open a chat it is empty since you can skip rendering any messages which are "NIL" (or whatever pattern you choose)

>  it would need to be part of the future releases as I've stated.

as always :)

**Comment by hantu85** [comment]
>> That is not the same and I do not suppose this would be expected result of chat history removal operation.

> Not sure a follow, in a federated system we should not expect the history to disappear from MAM or other devices.

What I mean, is removal is removal, so all details about the message exchange should be removed from the device if that is possible. Not only tombstoned or have body replaced, but actually removed with all metadata about the message being removed as well, ie. who sent message, when, etc. 

In other case, the name of this functionality should be "hide" instead of "delete" and I would prefer not to mislead users.

**Comment by licaon-kter** [comment]
Note that given the spread of a message (MAM, other devices, contacts) you'll always be "hiding" more or less.

Now, @fu3x11 what is your usecase (given the discussion above), local hide or?

**Comment by hantu85** [comment]
Question was about removal of data from the local device storage, not from the server.

**Comment by fu3x11** [comment]
> Note that given the spread of a message (MAM, other devices, contacts) you'll always be "hiding" more or less.
> 
> 
> 
> Now, @fu3x11 what is your usecase (given the discussion above), local hide or?

yes local hiding, on gajim, i have a setup where all messages expire after one week, server archival off, and i manually delete all chat history on that client when necessary, or for individual chats. (all locally obviously)

another cool feature gajim has, to prevent unauthorized group chat or dm leaks, is not trusting your contacts new omemo fingerprints by default, (it will be switched off completely), until you and your contact has consulted and verified that the new device was indeed him with one of many methods, then said fingerprint may be manually trusted

**Comment by leyloe** [comment]
> Question was about removal of data from the local device storage, not from the server.

messages can be archived on the server? let's just say your device gets seized by very bad people, so they have your keys, but there's no messages on the device. but the xmpp server stored your encrypted messages without your permission (with server side archival off on your account), and said server gave the bad guys the encrypted messages, would they be able to decrypt them? is there any mechanism in place to prevent that? does signal have a counter for that kind of attack
